utility/README.md

```diff
@@ -1,679 +1,1309 @@
+    A ->>+ WS: Delegate(GatewayPubKey)
+    WS ->>- A: ok
+```
 
+The following message will need to be signed by the Application's PrivateKey in order for it to be validated committed to the world state.
 
-```
-# ValidatorStakeMsg Interface
-type ValidatorStakeMsg interface {
-  GetPublicKey()  PublicKey       # The public cryptographic id of the custodial account
-  GetStakeAmount() BigInt         # The amount of uPOKT put in escrow, like a security deposit
-  GetServiceURL() ServiceURL      # The API endpoint where the validator service is provided
-  GetOperatorPubKey() PublicKey   # OPTIONAL; The non-custodial pubKey operating this node
+```go
+type DelegateMsg interface {
+  GetApplicationPublicKey() # The cryptographic ID of the Application
+  GetGatewayPublicKey() # The cryptographic ID of the Application
 }
 ```
 
+Once committed, the Application can be serviced on behalf of the Gateway. Though an Application can delegate to multiple Gateways simultaneously, the rate limiting for each session still remains at the Servicer level.
 
-Pocket Network 1.0 adopts a traditional Validator reward process such that only the Block Producer for each height is rewarded proportional to the total Transaction Fees + RelayRewards held in the produced block. Though the Block Producer selection mechanism is described in greater detail in the external Consensus Specification document, it is important to understand that the percentage of stake relative to the total stake is directly proportional to the likelihood of being the block producer. The net BlockProducerReward is derived from the TotalBlockReward by simply subtracting the DAOBlockReward amount. Once a proposal block is finalized into the blockchain, the BlockProducerReward amount of tokens is sent to the custodial account.
+#### 3.5.5 Application Servicing
 
+When an Application chooses to start a new session, the Gateway is responsible for dispatching the on-chain `StartSession` request and use an off-chain mechanism (e.g. AccessTokens) to service the Application. Throughout the duration of the session, validation and communication between the Application and Gateway are done using off-chain mechanisms, which are outside the scope of this document.
```

Comment:
What is on-chain StartSession request, is this referring to the dispatch RPC request or an actual transaction? It was inferred that sessions are implicitly “started” / generated based off a hash of variables as seen below:

“The seed data for a session is an implementation detail that could be composed of multiple variables. It includes, but is not limited to, attributes such as LatestBlockHash, Timestamp, etc…”